### PR TITLE
NodeAgent: Execute scripts in dedicated VM

### PR DESCRIPTION
**Pull request by jugglinmike**

### CODE REVIEW

**Comment by jugglinmike** [comment]
@bterlson I've added a test for cross-realm interactions as per your request in IRC. Note that it does not assert error names. This vagueness is intentional because not all hosts currently report the correct error, but that's a detail for Test262 to validate.

File: test/runify.js [hunk 1]
```diff
@@ -339,6 +339,52 @@ hosts.forEach(function (record) {
...
+                    return 1;
+                },
+                configurable: true
+              });
+
+              print('okay');
+          `).then(function(result) {
+             assert.equal(result.stderr, '');
+             assert(result.stdout.match(/^okay\r?\n/m));
+            });;
```
Comment (rwaldron) [inline test/runify.js]: nit: extra `;` and strange indent

File: test/runify.js [hunk 2]
```diff
@@ -339,6 +339,52 @@ hosts.forEach(function (record) {
...
+            assert(result.stdout.match(/^object\r?\nstring\r?\nnumber\r?\n/m));
+          });
+    });
+
+    it('observes correct cross-script interaction semantics', function () {
+      return agent.evalScript(`
+           print($.evalScript('let eshost;').type);
+           print($.evalScript('let eshost;').type);
+         `)
+        .then(function(result) {
```
Comment (rwaldron) [inline test/runify.js]: nit: all other tests are using arrow function syntax. This is not a blocker in any practical sense, but consistency is ideal.

File: test/runify.js (hunk 1 above)
Comment (jugglinmike) [inline test/runify.js]: Got it

File: test/runify.js (hunk 2 above)
Comment (jugglinmike) [inline test/runify.js]: The file is largely inconsistent in this regard, but I'm happy to use arrows here if you prefer.

**Comment by jugglinmike** [comment]
Thanks for the review, @rwaldron. It should be all set

**Comment by bterlson** [comment]
whoops, I conflicted this one with #30. @jugglinmike can you rebase?

**Comment by jugglinmike** [comment]
@bterlson Sure. Rebased on `master`, resolving conflicts and fixing up fixup commits.

(For posterity, [the original version of this branch is available here](https://github.com/bterlson/eshost/compare/master...bocoup:defensive-node-prebase)).

**Comment by rwaldron** [comment]
With the conflicts resolved, I will assume @bterlson is 👍

**Comment by jugglinmike** [comment]
Thank you both!

**Pull request merged**